### PR TITLE
secure supervisor endpoints

### PR DESCRIPTION
**Pull request by pjain1**

### CODE REVIEW

**Comment by himanshug** [comment]
:+1:

note: this PR adds following method to SupervisorSpec
`List<String> getDataSources();`

**Comment by fjy** [comment]
👍

File: server/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorSpec.java [hunk 1]
```diff
@@ -31,4 +33,6 @@
   String getId();
 
   Supervisor createSupervisor();
+
+  List<String> getDataSources();
```
Comment (jihoonson) [inline server/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorSpec.java]: @pjain1 @himanshug would you elaborate more on this? Do you have a supervisor indexing multiple dataSources? Would you tell me what use case needs that supervisor?

Comment (pjain1) [reply server/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorSpec.java]: don't remember exactly the reason for the list, may be it was for future proofing. If I am not wrong it was an ask from @himanshug, so may be he can explain better.

Comment (jihoonson) [reply server/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorSpec.java]: @pjain1 thanks. I'm implementing a new supervisor, and would like to add some javadoc to reduce confusion. Any comments will be much helpful.

**Pull request merged**